File: Assets/scenes/landinglaunchscene/script/AirlockEffectBehaviour.js
#pragma strict

private var particles : ParticleSystem;
private var lights : BlinkenFlareBehaviour[];
private var audioEffects : AudioSource[];



function Start () {
	particles = GetComponent(ParticleSystem);
	particles.enableEmission = false;
	
	lights = GetComponentsInChildren.<BlinkenFlareBehaviour>();
	
	audioEffects = GetComponentsInChildren.<AudioSource>();
}

//amount of air in the chamber, from 0 - 1
function setAtmosphereLevel(level : float){

	for( var s : AudioSource in audioEffects){
		s.volume = level;
	}

}

function Update () {

}

function start(){
	particles.enableEmission = true;
	for (var l : BlinkenFlareBehaviour in lights){
		l.blinking = true;
	}
	audioEffects[1].Play();
	
}

function stop(){
	particles.enableEmission = false;
	for (var l : BlinkenFlareBehaviour in lights){
		l.blinking = false;
	}
}

File: Assets/scenes/landinglaunchscene/script/AirlockEffectBehaviour.cs
using UnityEngine;
using System;


public class AirlockEffectBehaviour:MonoBehaviour{
	
	public ParticleSystem particles;
	public BlinkenFlareBehaviour[] lights;
	public AudioSource[] audioEffects;
	
	
	
	public void Start() {

		particles.enableEmission = false;
		

	}
	
	//amount of air in the chamber, from 0 - 1
	public void setAtmosphereLevel(float level){
	
		foreach(AudioSource s in audioEffects){
			s.volume = level;
		}
	
	}
	
	public void Update() {
	
	}
	
	public void start(){
		Debug.Log(gameObject.name);
		particles.enableEmission = true;
		foreach(BlinkenFlareBehaviour l in lights){
			l.blinking = true;
		}
		audioEffects[1].Play();
		
	}
	
	public void stop(){
		particles.enableEmission = false;
		foreach(BlinkenFlareBehaviour l in lights){
			l.blinking = false;
		}
	}
}